src/pages/index.tsx

```diff
@@ -2,128 +2,49 @@ import { FC, useState, useEffect, useContext } from 'react';
-const STATUS_ORDER = [
-  ASSIGNED,
-  COMPLETED,
-  BLOCKED,
-  AVAILABLE,
-  IN_PROGRESS,
-  SMOKE_TESTING,
-  NEEDS_REVIEW,
-  IN_REVIEW,
-  APPROVED,
-  MERGED,
-  SANITY_CHECK,
-  REGRESSION_CHECK,
-  RELEASED,
-  VERIFIED,
-];
-const statusActiveList = [
-  IN_PROGRESS,
-  BLOCKED,
-  SMOKE_TESTING,
-];
-async function updateCardContent(id: string, cardDetails: task) {
-  try {
-    const { requestPromise } = fetch({
-      url: `${TASKS_URL}/${id}`,
-      method: 'patch',
-      data: cardDetails,
-    });
-    await requestPromise;
-    toast(SUCCESS, 'Changes have been saved !');
-  } catch (err: any) {
-    if ('response' in err) {
-      toast(ERROR, err.response.data.message);
-      return;
-    }
-    toast(ERROR, err.message);
-  }
-}
-
 
 const Index: FC = () => {
-  const { state: appState } = useAppContext();  
-  const [filteredTask, setFilteredTask] = useState<any>([]);
+  const [filteredTasks, setFilteredTasks] = useState<null | task>(null);
   const { response, error, isLoading } = useFetch(TASKS_URL);
-  const { isEditMode } = appState;
-  const isUserAuthorized = useContext(isUserAuthorizedContext);
-  const isEditable = isUserAuthorized && isEditMode;
+
   useEffect(() => {
     if ('tasks' in response) {
-      const tasks = updateTasksStatus(response.tasks);
-      tasks.sort((a: task, b: task) => +a.endsOn - +b.endsOn);
-      tasks.sort((a: task, b: task) => STATUS_ORDER.indexOf(a.status)
-        - STATUS_ORDER.indexOf(b.status));
-      const taskMap: any = [];
-      tasks.forEach((item) => {
-        if (item.status in taskMap) {
-          taskMap[item.status] = [...taskMap[item.status], item];
-        } else {
-          taskMap[item.status] = [item];
-        }
-      });
-      setFilteredTask(taskMap);
+      const taskMap = getTaskMap(response.tasks);
+      if (Object.keys(taskMap)?.length) { // handles empty object
+        setFilteredTasks(taskMap);
+      }
     }
 
-    return(() => {
-      setFilteredTask([]);
+    return (() => {
+      setFilteredTasks(null);
     });
   }, [isLoading, response]);
 
   return (
     <Layout>
       <Head title='Tasks' />
-        <TasksProvider >
-          <div className={classNames.container}>
-            {!!error && <p>Something went wrong, please contact admin!</p>}
-            {isLoading ? (
-              <p>Loading...</p>
-            ) : (
-              <>
-                {Object.keys(filteredTask).length > 0
-                  ? Object.keys(filteredTask).map((key) => (
-                    <Accordion open={(statusActiveList.includes(key))} title={key} key={key}>
-                      <TaskList tasks={filteredTask[key]} isEditable={isEditable} updateCardContent={updateCardContent} hasLimit={key == IN_PROGRESS}/>
-                    </Accordion>
-                  ))
-                  : !error && 'No Tasks Found'}
-              </>
-            )}
-          </div>
+      <TasksProvider >
+        <div className={classNames.container}>
```

Comment:
This can also be done this way which will help in improving readability.
```jsx
if (error)
{
return <p>Something went wrong, please contact admin </p>
}

if(isLoading)
{
return 
<p>Loading...</p>
}

//This condition can be optional if Tabs component is showing message for empty data
if(isEmpty(filteredTasks)){
return <p>message</p>
}

return(
....
)
```